### PR TITLE
Github auth started

### PR DESCRIPTION
**Pull request by EnriqueVidal**
Rel #1

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,12 +4,16 @@ gem 'rails', '3.2.3'
 gem 'json'
 gem 'capistrano'
 gem 'haml-rails'
+gem 'system_timer'
+gem 'omniauth', :git => 'git://github.com/intridea/omniauth.git'
```
Comment (marioBonales) [inline Gemfile:8]: Any reason to use the git instead of the rubygems repository version?

Comment (EnriqueVidal) [inline Gemfile:8]: The released version fails, the problem has been fixed in master but the gem has not been released yet, I'll change this as soon as omniauth is released again.

Comment (marioBonales) [inline Gemfile:8]: Cool

**Comment by marioBonales** [comment]
I'm a little bit weirded out by the `dummy_controller`, Probably a placeholder anyways.

This is a bit of personal choice, but we should probably move `GITHUB_ID` and `GITHUB_SECRET` in a configuration file and provide an example, and a note on the readme.

Another thing! You're taking the public email from the profile, this means that if a user (me, for example) doesn't have a public email it will throw a validation exception `Validation failed: Email can't be blank`.

**Comment by EnriqueVidal** [comment]
@marioBonales `DummyController` is indeed a placeholder.

About `GITHUB_ID` and `GITHUB_SECRET` I put those in an initializer because that gets loaded once before the whole rails stack.

And about the token can you bring some light into this matter by providing your example `request` hash?

**Comment by marioBonales** [comment]
Here's the important snip:

```
provider: github
info: !map:OmniAuth::AuthHash::InfoHash 
  email: 
  nickname: marioBonales
  name: Mario Bonales
  urls: !map:Hashie::Mash 
    Blog: http://twitter.com/MarioBonales
    GitHub: https://github.com/marioBonales
uid: 940231
extra: !map:Hashie::Mash 
  raw_info: !map:Hashie::Mash 
    public_repos: 9
    followers: 4
    login: marioBonales
    email: 
    created_at: "2011-07-26T18:02:10Z"
    html_url: https://github.com/marioBonales
    bio: 
    url: https://api.github.com/users/marioBonales
    type: User
    following: 17
    blog: http://twitter.com/MarioBonales
    hireable: false
    gravatar_id: 3d2786333841a3a6f43dc88214caef38
    avatar_url: https://secure.gravatar.com/avatar/3d2786333841a3a6f43dc88214caef38?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-140.png
    company: 
    name: Mario Bonales
    id: 940231
    public_gists: 5
    location: "Tijuana, M\xC3\xA9xico"
```

email is nil because is trying to get the public email in my profile (the one you put in https://github.com/settings/profile)

**Comment by EnriqueVidal** [comment]
I'd really like to have their emails to later notify them about meetings, can you figure out for us aside from the email what else can an user hide?

**Comment by marioBonales** [comment]
Here's a method in the github api that gives you the private emails that the user might have: http://developer.github.com/v3/users/emails/

Also, You don't have to have a `name` but I think that's not being validated with us, so cool

**Comment by EnriqueVidal** [comment]
Hmmm... too intrusive, I think we better drop the email validation and give them a way to update their registration later on if they wish to be notified, are you cool with that change?

**Comment by marioBonales** [comment]
Sure, you could provide an email field if they don't have their email address public in github

**Comment by EnriqueVidal** [comment]
Ok, for now I'll remove the email validation, we can discuss the registration controller and views in a different issue, I will have to drop the uniqueness of my `email` field.

**Comment by marioBonales** [comment]
That's good

**Pull request merged**